tests/unit/test_project_structure.py

```diff
@@ -0,0 +1,153 @@
+from unittest.mock import patch
+import datetime as dt
+import time
+
+import pandas as pd
+from lightwood.api import dtype
+
+from mindsdb_sql import parse_sql
+
+from .executor_test_base import BaseExecutorDummyML
+
+
+class TestProjectStructure(BaseExecutorDummyML):
+
+    def wait_predictor(self, project, name, filter=None):
+        # wait
+        done = False
+        for attempt in range(200):
+            sql = f"select * from {project}.models_versions where name='{name}'"
+            if filter is not None:
+                for k, v in filter.items():
+                    sql += f" and {k}='{v}'"
+            ret = self.run_sql(sql)
+            if not ret.empty:
+                if ret['STATUS'][0] == 'complete':
+                    done = True
+                    break
+                elif ret['STATUS'][0] == 'error':
+                    break
+            time.sleep(0.5)
```

Comment:
Do not use sleep in tests it is an anti-pattern which produces intermittent behavior. If you need wait until something happen, check this condition explicitly.

Reply:
why is anti-pattern? (and what is name if this anti-pattern?) 
we need to wait because predictor is training in subprocess. sleep is saving process time

Reply:
`join_learn_process` should be solution to not wait explicitly :)